## Calculating top-1 and top-5 score

In [9]:
from top_n import get_top_n_score

#### Top-1 and Top-5 score for graph: (vlad embeddings (3 to 5) and objects (4, 6))

In [20]:
top1_score_test1, top5_score_test1 = get_top_n_score("test_set_graph_vlad_object_embeddings.txt")

print('Top-1 score: ', top1_score_test1)
print('Top-5 score: ', top5_score_test1)

Top-1 score:  0.475
Top-5 score:  0.7833333333333333


#### Top-1 and Top-5 score for vlad embeddings taken from scans

In [29]:
top1_score_test2, top5_score_test2 = get_top_n_score("test_set_vlad_embeddings.txt")
print('Top-1 score: ', top1_score_test2)
print('Top-5 score: ', top5_score_test2)

Top-1 score:  0.23505154639175257
Top-5 score:  0.45979381443298967


#### Top-1 and Top-5 score for graph: (vlad embeddings(7 to 10))

In [30]:
top1_score_test3, top5_score_test3 = get_top_n_score("test_set_vlad_embeddings_only_in_graph.txt")
print('Top-1 score: ', top1_score_test3)
print('Top-5 score: ', top5_score_test3)

Top-1 score:  0.48541666666666666
Top-5 score:  0.7770833333333333


#### Top-1 and Top-5 score for graph: (objects (7 to 10))

In [31]:
top1_score_test4, top5_score_test4 = get_top_n_score("test_set_objects_only.txt")
print('Top-1 score: ', top1_score_test4)
print('Top-5 score: ', top5_score_test4)

Top-1 score:  0.4791666666666667
Top-5 score:  0.74375


#### Top-1 and Top-5 Score for graph (all vlad embeddings, all objects): 

In [32]:
top1_score_test5, top5_score_test5 = get_top_n_score("test_set_all_objects_all_vlad_embeddings.txt")
print('Top-1 score: ', top1_score_test5)
print('Top-5 score: ', top5_score_test5)                    

Top-1 score:  0.5104166666666666
Top-5 score:  0.7729166666666667


#### Conclusion: there is no significance difference between different strategies of objects and embeddings inclusions, therefore, we should work on other methods of graph connectivity. However, we showed that graph embeddings represent much higher prediction score compared to vlad embeddings (0.783 compared to 0.453 among top-n accuracy scores)